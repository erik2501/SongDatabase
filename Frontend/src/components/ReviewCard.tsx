import Card from "@mui/material/Card";
import Rating from "@mui/material/Rating";
import StarIcon from '@mui/icons-material/Star';
import internal from "stream";

interface reviewProp {
    userName: string,
    star: number,
    description: string,
}

export default function ReviewCard(props: reviewProp) {
    return (
        <div>
            <Card variant="outlined">
                <h5>{props.userName}</h5>
                <Rating name="read-only" value={props.star} readOnly emptyIcon={<StarIcon fontSize="inherit" />}/>
                <p>{props.description}</p>
            </Card>
        </div>
    )
}