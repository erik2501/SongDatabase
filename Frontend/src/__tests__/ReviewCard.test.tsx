import ReviewCard from "../components/ReviewCard"
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react';


it('renders when passed name and rating and description', () => {
    const tree = renderer.create((<ReviewCard userName='snapshotTestName' star={1} description="bra" />)).toJSON();
expect(tree).toMatchSnapshot();
});

