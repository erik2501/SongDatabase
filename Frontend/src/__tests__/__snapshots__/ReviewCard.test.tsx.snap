// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when passed name and rating and description 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    style={
      Object {
        "paddingBottom": "15px",
      }
    }
  >
    <h5>
      snapshotTestName
    </h5>
    <span
      aria-label="1 Star"
      className="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role="img"
    >
      <span>
        <span
          className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span>
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <p>
      bra
    </p>
  </div>
</div>
`;
